src/components/common/Members/Members.tsx

```diff
@@ -81,19 +83,42 @@ const Members = ({ selectedDomain, handleDomainChange, filters }: Props) => {
     fetchPolicy: 'cache-and-network',
   });
 
-  const contributors =
-    data?.getMembersForColony?.contributors?.filter(notNull) ?? [];
-  const watchers = data?.getMembersForColony?.watchers?.filter(notNull) ?? [];
+  useEffect(() => {
+    setContributors(
+      data?.getMembersForColony?.contributors?.filter(notNull) ?? [],
+    );
+    setWatchers(data?.getMembersForColony?.watchers?.filter(notNull) ?? []);
+  }, [data]);
+
+  const filterContributorsAndWatchers = useCallback(() => {
+    const filteredContributors = filterMembers<Contributor>(
+      data?.getMembersForColony?.contributors || [],
+      searchValue,
+      filters,
+    );
+    setContributors(filteredContributors);
+    const filteredWatchers = filterMembers<Watcher>(
+      data?.getMembersForColony?.watchers || [],
+      searchValue,
+      filters,
+    );
+    setWatchers(filteredWatchers);
+  }, [data, filters, searchValue]);
 
   // handles search values & close button
   const handleSearch = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       const value = event.target?.value || '';
       setSearchValue(value);
+      filterContributorsAndWatchers();
     },
-    [setSearchValue],
+    [filterContributorsAndWatchers, setSearchValue],
   );
 
+  useEffect(() => {
```

Comment:
@willm30 I agree, especially having read the docs page you shared a few weeks ago: https://beta.reactjs.org/learn/you-might-not-need-an-effect

I don't think we need a state for members/contributors at all since they're derived from state (`data`, `searchValue`) and props (`filters`).

We could just call the `filterMembers` directly in the component body:
```js
const filteredContributors = filterMembers<Contributor>(
      data?.getMembersForColony?.contributors || [],
      searchValue,
      filters,
);
```

Or, if it proves to be an expensive computation, wrap it in `useMemo` with `data`, `searchValue` and `filters` as dependencies.